impl/src/main/java/org/ehcache/internal/store/tiering/CacheStore.java

```diff
@@ -0,0 +1,350 @@
+  }
+
+  @Override
+  public void init() {
+    cachingTier.init();
+    authoritativeTier.init();
+  }
+
+  @Override
+  public void maintenance() {
+    cachingTier.maintenance();
+    authoritativeTier.maintenance();
+  }
+
+  @Override
+  public void enableStoreEventNotifications(StoreEventListener<K, V> listener) {
+    authoritativeTier.enableStoreEventNotifications(listener);
+  }
+
+  @Override
+  public void disableStoreEventNotifications() {
+    authoritativeTier.disableStoreEventNotifications();
+  }
+
+  @Override
+  public Iterator<Cache.Entry<K, ValueHolder<V>>> iterator() throws CacheAccessException {
+    return authoritativeTier.iterator();
+  }
+
+  @Override
+  public ValueHolder<V> compute(final K key, final BiFunction<? super K, ? super V, ? extends V> mappingFunction) throws CacheAccessException {
+    try {
+      return authoritativeTier.compute(key, mappingFunction);
+    } finally {
+      cachingTier.remove(key);
+    }
+  }
+
+  @Override
+  public ValueHolder<V> compute(final K key, final BiFunction<? super K, ? super V, ? extends V> mappingFunction, final NullaryFunction<Boolean> replaceEqual) throws CacheAccessException {
+    try {
+      return authoritativeTier.compute(key, mappingFunction, replaceEqual);
+    } finally {
+      cachingTier.remove(key);
+    }
+  }
+
+  public ValueHolder<V> computeIfAbsent(final K key, final Function<? super K, ? extends V> mappingFunction) throws CacheAccessException {
+    try {
+      return cachingTier.getOrComputeIfAbsent(key, new Function<K, ValueHolder<V>>() {
+        @Override
+        public ValueHolder<V> apply(K k) {
+          try {
+            return authoritativeTier.computeIfAbsentAndFault(k, mappingFunction);
+          } catch (CacheAccessException cae) {
+            throw new ComputationException(cae);
+          }
+        }
+      });
+    } catch (ComputationException ce) {
+      throw ce.getCacheAccessException();
+    }
+  }
+
+  @Override
+  public ValueHolder<V> computeIfPresent(final K key, final BiFunction<? super K, ? super V, ? extends V> remappingFunction) throws CacheAccessException {
+    try {
+      return authoritativeTier.computeIfPresent(key, remappingFunction);
+    } finally {
+      // XXX: it's not possible to discriminate if computation has been performed or not by looking at the computeIfPresent return value
+      // Hence, we must always clear the caching tier even when that's not necessary (ie: when no computing happened)
+      cachingTier.remove(key);
+    }
+  }
+
+  @Override
+  public ValueHolder<V> computeIfPresent(final K key, final BiFunction<? super K, ? super V, ? extends V> remappingFunction, final NullaryFunction<Boolean> replaceEqual) throws CacheAccessException {
+    try {
+      return authoritativeTier.computeIfPresent(key, remappingFunction, replaceEqual);
+    } finally {
+      // XXX: it's not possible to discriminate if computation has been performed or not by looking at the computeIfPresent return value
```

Comment:
Do these two `XXX` imply you want to try to do something smarter about it? If not, I'd just leave the comment in, without it... Not sure what all people mean with that prefix in comments